core/event/task/task.go

```diff
@@ -37,6 +37,13 @@ func Once(task Task) Task {
 	}
 }
 
+func Delay(task Task, duration time.Duration) Task {
+	return func(ctx context.Context) error {
+		time.Sleep(duration)
+		return task(ctx)
```

Comment:
This needs to be on another goroutine, as it would otherwise block. I think it makes sense to do it here.